templates/daemonset.yaml

```diff
@@ -50,6 +50,38 @@ spec:
             runAsUser: 0
           resources:
 {{ toYaml .Values.sysctlImage.resources | indent 12 }}
+      {{- end }}
+        {{- if .Values.slave.persistUniqueID }}
+        - name: init-nodeuid
+          image: busybox
+          volumeMounts:
+            - name: nodeuid
+              mountPath: "/nodeuid"
+          env:
+            - name: MY_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          command:
+            - "/bin/sh"
+          args:
+            - "-c"
+            - '
+            TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token);
+            URL="https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}/api/v1/nodes/${MY_NODE_NAME}";
+            HEADER="Authorization: Bearer ${TOKEN}";
+
+            DATA=$(wget -q -T 5 --no-check-certificate --header "${HEADER}" -O - "${URL}");
+            [ -z "${DATA}" ] && exit 1;
+
+            UID=$(echo "${DATA}" | grep -m 1 systemUUID | grep -o ":.*" | tr -d ": \",");
```

Comment:
[Spec says](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/#nodesysteminfo-v1-core) `machineID` is _preferred_ as method of identifying unique machines in a cluster. The spec is all I read though, so not sure how reliable `systemUUID` is.

Reply:
i saw, `machineID` is changed on reboot.

Reply:
@ncmans @varyumin would it be possible to have same `systemUUID` if nodes are VMs running on the same host?

example: [link](https://github.com/kubernetes-sigs/kind/issues/1027)

Probably we could make this filed configurable and let user decide and use `uid` by default?

Reply:
`uid` is 100% safe and covers almost all cases.

Reply:
are you talking about [`metdata.uid`](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids) ?

Reply:
yes

Reply:
https://github.com/netdata/helmchart/issues/61#issuecomment-608362343

Reply:
Are you sure that this should be done in the Helm Chart with the help of scripts?

@ilyam8  How did you test when `machineID` changed?

Reply:
> How did you test when machineID changed?

I am not sure. I have no physical node to reboot. As i said i restarted vbox  vm. I understand that is not good test. But i have what i have atm 😄 

> Are you sure that this should be done in the Helm Chart with the help of scripts?

What do you mean? My opinion on implementation

> https://github.com/netdata/helmchart/issues/61#issuecomment-607944497

Reply:
I mean like this https://github.com/varyumin/netdata_uid. I have written on Golang small application. And can be better this will add to main repository netdata and use this binary file inside netdata docker. It's my opinion )))) It may be wrong)

Reply:
It is not better imo. I would prefer to have a docker image and use it instead of bash code in args, but it doesnt worth it atm. We got what we need, implementation is details, it doesnt couple netdata with implementation and is very likely we will change it in the future.